includes/panopto_derivatives.inc

```diff
@@ -5,37 +5,356 @@
+        
+        islandora_video_streaming_panopto_create_manifest_for_video($filePath, MANIFEST_FILE_NAME);
+        islandora_video_streaming_panopto_upload_file($upload_target, MANIFEST_FILE_NAME);
+        islandora_video_streaming_panopto_upload_file($upload_target, $filePath);
+        islandora_video_streaming_panopto_finish_upload($uploadendpoint, $session, $token);
+        
+        try {
+          $success = monitor_progress($uploadendpoint, $upload_id, $token);
+        } catch (Exception $ex) {
+          $messages = array(
+          'message' => t('Panopto progress error: @exception'),
+          'message_sub' => array('@exception' => $ex->getMessage()),
+          'type' => 'watchdog',
+          'severity' => WATCHDOG_ERROR,
+          );
+          $failureMessage['messages'][] = $messages;
+          watchdog_exception('islandora_video_streaming_panopto', $ex);
+          return $failureMessage;
+        }
+        $streaming['upload_id'] = $session['ID'];
+      } catch (Exception $ex) {
+        $messages = array(
+          'message' => t('Panopto upload error: @exception'),
+          'message_sub' => array('@exception' => $ex->getMessage()),
+          'type' => 'watchdog',
+          'severity' => WATCHDOG_ERROR,
+        );
+        $failureMessage['messages'][] = $messages;
+        watchdog_exception('islandora_video_streaming_panopto', $ex);
+      }
+    } else {
+      $messages = array(
+        'message' => t('Failed to retrieve session for object: @pid'),
+        'message_sub' => array('@pid' => $object->id),
+        'type' => 'watchdog',
+        'severity' => WATCHDOG_ERROR,
+      );
+      $failureMessage['messages'][] = $messages;
+      return $failureMessage;
+    }
+    file_delete($archival_file['file'], true);
     if ($streaming) {
        // Write the data to a temporary file.
       $temp_name = drupal_tempnam('temporary://', 'file');
       if (file_put_contents($temp_name, json_encode($streaming)) === FALSE) {
-        $failureMessage['messages'][0]['message'] = t('Failed to create temporary file @filename for object @pid when uploading to Panopto.');
-        $failureMessage['messages'][0]['message_sub']['@filename'] = $temp_name;
+        $messages = array(
+            'message' => t("Failed to create temporary file @filename for object @pid when uploading to Panopto"),
+            'message_sub' => array(
+              '@pid' => $object->id, 
+              '@filename' => $temp_name
+            ),
+            'type' => 'watchdog',
+            'severity' => WATCHDOG_ERROR,
+        );
+        $failureMessage['messages'][] = $messages;
         return $failureMessage;
       }
-      return islandora_video_add_datastream($object, 'KALTURA_STREAM', $temp_name);
+      return islandora_video_add_datastream($object, 'PANOPTO_INFO', $temp_name);
     } else {
       return $failureMessage;
     }
   }
 }
 
+/*
+ * Requests token and returns token
+ * @param String $oauthendpoint 
```

Comment:
I think the typing of `$oauthendpoint` should be `string` rather than `String`.  `String` represents a PHP Object, `string` represents the primitive type.  Add a human readable description of what this variable should contain, e.g.: "The URL to the Panopto OAuth connect/token endpoint".

Please add the PHPDoc for the `@return` for `islandora_video_streaming_panopto_get_token`.

Reply:
Done and will check/fix documentation for other functions as well.